### PR TITLE
feature - home screen

### PR DESCRIPTION
**Pull request by at-phatnguyen**
doing login screen

### CODE REVIEW

File: MyApp/View/Base/AppDelegate.swift [hunk 1]
```diff
@@ -16,6 +16,7 @@ let networkIndicator = NetworkActivityIndicatorManager.shared
 final class AppDelegate: UIResponder, UIApplicationDelegate {
 
     var window: UIWindow?
+
```
Comment (at-daiho) [inline MyApp/View/Base/AppDelegate.swift]: WARNING - Limit vertical whitespace to a single empty line. Currently 2.

File: MyApp/View/Base/AppDelegate.swift [hunk 2]
```diff
@@ -16,6 +16,7 @@ let networkIndicator = NetworkActivityIndicatorManager.shared
...
```
Comment (at-daiho) [inline MyApp/View/Base/AppDelegate.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Base/AppDelegate.swift [hunk 3]
```diff
@@ -27,6 +28,14 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         configNetwork()
         configDatabase()
+
```
Comment (at-daiho) [inline MyApp/View/Base/AppDelegate.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Base/AppDelegate.swift [hunk 4]
```diff
@@ -27,6 +28,14 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         configNetwork()
         configDatabase()
+        
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.backgroundColor = UIColor.white
+        window?.makeKeyAndVisible()
+
```
Comment (at-daiho) [inline MyApp/View/Base/AppDelegate.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/View/Base/AppDelegate.swift [hunk 5]
```diff
@@ -27,6 +28,14 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
...
         configNetwork()
         configDatabase()
+        
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.backgroundColor = UIColor.white
+        window?.makeKeyAndVisible()
+        
+        let vc = HomeViewController()
+        window?.rootViewController = vc
+
```
Comment (at-daiho) [inline MyApp/View/Base/AppDelegate.swift]: WARNING - Lines should not have trailing whitespace.

**Comment by at-daiho** [comment]
Total linter violations in pull request: 5
Serious: 0
Warnings: 5

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 6]
```diff
@@ -7,10 +7,53 @@
...
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel.numberOfSections()
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.numberOfRow(at: section)
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        return viewModel.cellForRow(tableView: tableView, indexPath: indexPath)
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: cái ni sai rồi!

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 7]
```diff
@@ -7,10 +7,53 @@
...
+    }
+
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return titleHeader[section]
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: viewmodel

File: MyApp/ViewModel/HomeViewModel/HomeViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+//
+//  Created by MAC-PHAT on 7/25/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import UIKit
+import MVVM
+
+class HomeViewModel: MVVM.ViewModel {
+    var indexPath = IndexPath()
```
Comment (at-tuanmai2) [inline MyApp/ViewModel/HomeViewModel/HomeViewModel.swift]: ?

File: MyApp/ViewModel/HomeViewModel/HomeViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+    let  slideImages: [String] = ["1", "2", "3", "4"]
```
Comment (at-tuanmai2) [inline MyApp/ViewModel/HomeViewModel/HomeViewModel.swift]: demo file

File: MyApp/ViewModel/HomeViewModel/HomeViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    func heightOfSections(indexPath: IndexPath) -> CGFloat {
+        if indexPath.section == 0 {
+            return 100
+        }
+        return 200
+    }
+
+    func cellForRow(tableView: UITableView, indexPath: IndexPath) -> UITableViewCell {
```
Comment (at-tuanmai2) [inline MyApp/ViewModel/HomeViewModel/HomeViewModel.swift]: ko cần!

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 11]
```diff
@@ -7,10 +7,61 @@
...
+    private var indexItem: Int = 0
+    var viewModel = HomeViewModel()
     override func viewDidLoad() {
         super.viewDidLoad()
+        configTable()
+    }
+
+    func configTable() {
+        let home = UIBarButtonItem(title: "Home", style: .plain, target: self, action: #selector(HomeViewModel.didSelectHome))
+        self.navigationItem.leftBarButtonItem = home
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: remove `self`

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 12]
```diff
@@ -7,10 +7,61 @@
...
+    @IBOutlet weak var tableView: UITableView!
+    private var indexItem: Int = 0
+    var viewModel = HomeViewModel()
     override func viewDidLoad() {
         super.viewDidLoad()
+        configTable()
+    }
+
+    func configTable() {
+        let home = UIBarButtonItem(title: "Home", style: .plain, target: self, action: #selector(HomeViewModel.didSelectHome))
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: Define name

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 13]
```diff
@@ -7,10 +7,61 @@
...
+        self.navigationItem.leftBarButtonItem = home
+        let search = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: #selector(HomeViewModel.didSelectSearchBarItem))
+        self.navigationItem.rightBarButtonItem = search
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: remove `self`

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 14]
```diff
@@ -7,10 +7,61 @@
...
+        return viewModel.numberOfSections()
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.numberOfRow(at: section)
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.section == 0 {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "TrendHomeCell", for: indexPath) as? TrendHomeCell else {
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: register use `Swift Utils` so In this use `Swift util` too.

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 15]
```diff
@@ -7,10 +7,61 @@
...
+                return UITableViewCell()
+            }
+            return cell
+        }
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "HomeCell", for: indexPath) as? HomeCell else {
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: register use `Swift Utils` so In this use `Swift util` too.

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 16]
```diff
@@ -7,10 +7,61 @@
...
+    }
+
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return viewModel.titleForHeaderInSection(tableView: tableView, section: section)
+    }
+}
+
+extension HomeViewController: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: Why it in delegate?

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 17]
```diff
@@ -7,10 +7,61 @@
...
+        return viewModel.heightOfSections(indexPath: indexPath)
```
Comment (at-truongnguyen) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: Check your function name `heightForRowAt` but return `heightOfSections`

File: MyApp/ViewModel/HomeViewModel/HomeViewModel.swift [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    func heightOfSections(indexPath: IndexPath) -> CGFloat {
+        if indexPath.section == 0 {
+            return 100
+        }
+        return 200
+    }
+
+        @objc func didSelectSearchBarItem() {
```
Comment (at-truongnguyen) [inline MyApp/ViewModel/HomeViewModel/HomeViewModel.swift]: what is this?

File: MyApp/ViewModel/HomeViewModel/HomeViewModel.swift [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+    // search someThing
+    }
+    @objc func didSelectHome() {
```
Comment (at-truongnguyen) [inline MyApp/ViewModel/HomeViewModel/HomeViewModel.swift]: what is this?

**Pull request merged**